### PR TITLE
Add Global Cache OLAP telemetry

### PR DESCRIPTION
**Pull request by kwapik**
### Description of change
This adds Global Cache metrics to OLAP telemetry.

##### Checklist

- [x] Tested in playground or other setup

## Summary by CodeRabbit

- **Refactor**
  - Renamed cache status labels for clarity and consistency.
  - Improved handling of cache lookup and operation statuses in telemetry data.

- **New Features**
  - Introduced new telemetry labels for global cache lookup and operation statuses.

- **Documentation**
  - Updated the observability documentation to reflect new and modified telemetry fields.

- **Bug Fixes**
  - Corrected capitalization in the `response_received` field.
  - Removed an outdated reject reason from the `aperture.reject_reason` field.

- **Style**
  - Standardized formatting in the `Flow` interface and related methods for better readability.

### CODE REVIEW

**Review by harjotgill — APPROVED** [review]

File: pkg/otelcollector/consts/consts.go [hunk 1]
```diff
@@ -87,10 +87,14 @@ const (
...
-	// ApertureCacheOperationStatusLabel describes status of the cache operation.
-	ApertureCacheOperationStatusLabel = "aperture.cache_operation_status"
+	// ApertureResultCacheLookupStatusLabel describes status of the cache lookup.
+	ApertureResultCacheLookupStatusLabel = "aperture.cache_lookup_status"
+	// ApertureResultCacheOperationStatusLabel describes status of the cache operation.
+	ApertureResultCacheOperationStatusLabel = "aperture.cache_operation_status"
+	// ApertureGlobalCacheLookupStatusLabel describes status of the cache lookup.
+	ApertureGlobalCacheLookupStatusLabel = "aperture.global_cache_lookup_status"
+	// ApertureGlobalCacheOperationStatusLabel describes status of the cache operation.
+	ApertureGlobalCacheOperationStatusLabel = "aperture.global_cache_operation_status"
```
Comment (harjotgill) [inline pkg/otelcollector/consts/consts.go:97]: @coderabbitai have these metrics been documented properly?

Comment (harjotgill) [reply pkg/otelcollector/consts/consts.go:97]: Did you search all markdown files?

Comment (harjotgill) [reply pkg/otelcollector/consts/consts.go:97]: Why are you searching for the variable names? We have to search the labels.

File: pkg/otelcollector/metricsprocessor/internal/check_response_labels.go
Comment (harjotgill) [inline pkg/otelcollector/metricsprocessor/internal/check_response_labels.go:1]: @coderabbitai provide me insights into the repository. Top committers, activity by day and authors, top modified files and so on. Crunch some numbers and make it an interesting report with tables and charts.

File: pkg/otelcollector/consts/consts.go (hunk 1 above)
Comment (harjotgill) [reply pkg/otelcollector/consts/consts.go:97]: Can you guide me where the documentation changes have to be made for the metrics?

Comment (harjotgill) [reply pkg/otelcollector/consts/consts.go:97]: Can you add the documentation in that file and provide me the diff?

File: pkg/otelcollector/metricsprocessor/internal/check_response_labels.go
Comment (harjotgill) [reply pkg/otelcollector/metricsprocessor/internal/check_response_labels.go:1]: Why is the author list not available? Figure it out and return the correct report.

Comment (harjotgill) [reply pkg/otelcollector/metricsprocessor/internal/check_response_labels.go:1]: Yes, please.

Comment (harjotgill) [reply pkg/otelcollector/metricsprocessor/internal/check_response_labels.go:1]: @coderabbitai: yes please

File: pkg/otelcollector/consts/consts.go (hunk 1 above)
Comment (kwapik) [reply pkg/otelcollector/consts/consts.go:97]: @coderabbit This is not correct place. These are flow metrics, not prometheus metrics.

**Pull request merged**